### PR TITLE
webhooks: Handle known processing errors

### PR DESCRIPTION
**Pull request by elie222**
Handle expected webhook processing failures without forwarding them to error reporting.

- Stop structured safe errors at the shared webhook boundary.
- Preserve propagation for unexpected provider and network failures.
- Add regression coverage for known errors.

Tests: pnpm test utils/webhook/process-history-item.test.ts; pnpm test utils/webhook/google/process-history.test.ts utils/webhook/outlook/process-history.test.ts

<a href="https://cubic.dev/pr/elie222/inbox-zero/pull/3103?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/web/utils/webhook/process-history-item.test.ts [hunk 1]
```diff
@@ -146,6 +148,34 @@ describe("Provider Edge Cases", () => {
...
+    it("handles safe errors without forwarding them to webhook error handlers", async () => {
+      const provider = createMockEmailProvider({
+        getMessage: vi.fn().mockResolvedValue(
+          getMockParsedMessage({
+            labelIds: ["INBOX"],
+          }),
+        ),
+        isSentMessage: vi.fn().mockReturnValue(false),
+      });
+      vi.mocked(runRules).mockRejectedValueOnce(
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/webhook/process-history-item.test.ts:161]: Added an explicit assertion that rule processing is reached, so the regression cannot pass through an unrelated early return. The focused webhook test suite passes.
